e2etests/test/testUtils.js

```diff
@@ -82,8 +82,24 @@ function closeDeviceEventHubClients(deviceClient, eventHubClient, ehReceivers, d
   }
 }
 
+function getErrorDetailString(functionName, err) {
+  var detail = functionName + ' returned ' + (err ? err : 'success');
+  if (err && err.responseBody) {
+    if (err.response && err.response.headers) {
+      if (err.response.headers['content-type'].indexOf('json') !== -1) {
+        detail += '\n';
+        var body = JSON.parse(err.responseBody);
+        delete body.StackTrace;
+        detail += JSON.stringify(body, null, ' ');
+      }
```

Comment:
you could also enclose the JSON.parse in a try catch statement that catches SyntaxError and not worry about parsing headers